api/runtime/hooks/v1alpha1/inplaceupdate_types.go

```diff
@@ -134,21 +136,19 @@ type CanUpdateMachineSetResponse struct {
 	// CommonResponse contains Status and Message fields common to all response types.
 	CommonResponse `json:",inline"`
 
-	// machineSetPatch indicates MachineSet spec changes handled in-place.
+	// machineSetPatch when applied to the current MachineSet spec, indicates changes handled in-place.
 	// +optional
-	MachineSetPatch *Patch `json:"machineSetPatch,omitempty"`
+	MachineSetPatch Patch `json:"machineSetPatch,omitempty,omitzero"`
 
-	// infrastructureMachineTemplateSpecPatch indicates infra template spec changes handled in-place.
+	// infrastructureMachineTemplatePatch indicates infra template spec changes handled in-place.
```

Comment:
Just consistently dropping Spec from the field names here (the patches should only contain spec changes as documented, I just wouldn't surface it in field names)